packages/app/src/contacts/services/contact.service.ts

```diff
@@ -150,16 +150,42 @@ export class ContactService {
       return isContactValidResponse
     }
 
-    const { ok, data } = await this.deviceManager.device.request({
+    const { ok, data, error } = await this.deviceManager.device.request({
       endpoint: Endpoint.Contacts,
       method: Method.Put,
       body: ContactPresenter.mapToPureContact(contact),
     })
 
     if (ok) {
       this.contactRepository.update(contact, true)
+      return {
+        status: RequestResponseStatus.Ok,
+        data: contact,
+      }
+      // error type cannot be typed correctly, response method needs enhancement
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    } else if (error?.payload?.status === "phone-number-duplicated") {
```

Comment:
Each conditional statement contains "return", it could be easier to read with only "if" statements instead of "else if".